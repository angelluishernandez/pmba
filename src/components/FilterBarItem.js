import React from "react";
import { FilterBarItemWrapper } from "./styled-components/FilterBarItemWrapper";

const FilterBarItem = ({ genre, handleRadioChange, filtersSelected }) => {
	const isChecked = filtersSelected.includes(genre.toLowerCase());

	return (
		<FilterBarItemWrapper>
			<div>
				<label htmlFor={genre}>{genre}</label>
				<input
					type="radio"
					checked={isChecked}
					value={genre}
					onChange={(e) => handleRadioChange(e)}
				/>
			</div>
		</FilterBarItemWrapper>
	);
};

export default FilterBarItem;
