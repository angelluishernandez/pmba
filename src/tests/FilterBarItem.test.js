import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import React from "react";

import FilterBarItem from "../components/FilterBarItem";

configure({ adapter: new Adapter() });

test("should render FilterBarItem correctly", () => {
	const genre = "genre";
	const filtersSelected = [];
	const wrapper = shallow(
		<FilterBarItem filtersSelected={filtersSelected} genre={genre} />
	);
	expect(toJSON(wrapper)).toMatchSnapshot();
});
