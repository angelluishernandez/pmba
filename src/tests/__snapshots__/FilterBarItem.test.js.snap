// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterBarItem correctly 1`] = `
<styled.div>
  <label
    htmlFor="genre"
  >
    genre
  </label>
  <input
    checked={false}
    onChange={[Function]}
    type="radio"
    value="genre"
  />
</styled.div>
`;
